File: src/main/java/org/kite9/diagram/common/elements/mapping/BaseGridCornerVertices.java
package org.kite9.diagram.common.elements.mapping;

import org.kite9.diagram.model.DiagramElement;

public class BaseGridCornerVertices extends AbstractBaseCornerVertices {

	public BaseGridCornerVertices(DiagramElement container, int depth) {
		super(container, FULL_RANGE, FULL_RANGE, depth);
		createInitialVertices(null);
	}

	@Override
	protected String getVertexIDStem() {
		return super.getVertexIDStem()+"-g";
	}

	public DiagramElement getGridContainer() {
		return rootContainer;
	}

	
}

File: visualization/kite9-visualization-common/src/jvmMain/kotlin/org/kite9/diagram/common/elements/mapping/BaseGridCornerVertices.kt
package org.kite9.diagram.common.elements.mapping

import org.kite9.diagram.model.DiagramElement

class BaseGridCornerVertices(container: DiagramElement, depth: Int) :
    AbstractBaseCornerVertices(container, FULL_RANGE, FULL_RANGE, depth) {

    override fun getVertexIDStem(): String {
        return super.getVertexIDStem() + "-g"
    }

    fun getGridContainer(): DiagramElement {
        return rootContainer
    }

    init {
        createInitialVertices(null)
    }
}